dash-spv-coinjoin/src/messages/coinjoin_broadcast_tx.rs

```diff
@@ -0,0 +1,61 @@
+use std::io;
+use std::io::{Error, Write};
+use dash_spv_masternode_processor::crypto::UInt256;
+use dash_spv_masternode_processor::consensus::{Decodable, Encodable, encode};
+use dash_spv_masternode_processor::tx::Transaction;
+
+// dstx
+#[repr(C)]
+#[derive(Clone, Debug)]
+#[ferment_macro::export]
+pub struct CoinJoinBroadcastTx {
+    pub tx: Transaction,
+    // pub masternode_outpoint: TransactionOutPoint, // TODO: versioning
+    pub pro_tx_hash: UInt256,
+    pub signature: Option<Vec<u8>>,
+    pub signature_time: i64,
+}
+
+impl Encodable for CoinJoinBroadcastTx {
+    #[inline]
+    fn consensus_encode<W: Write>(&self, mut writer: W) -> Result<usize, Error> {
+        let mut offset = 0;
+        let tx_data = self.tx.to_data(); // TODO: consensus_encode
+        writer.write_all(&tx_data)?;
+        offset += tx_data.len();
+
+        // TODO: versioning
+        // if (protocolVersion >= params.getProtocolVersionNum(NetworkParameters.ProtocolVersion.COINJOIN_PROTX_HASH)) {
+            offset += self.pro_tx_hash.consensus_encode(&mut writer)?;
+        // } else {
+        //     masternodeOutpoint.bitcoinSerialize(stream);
+        // }
```

Comment:
This is another place were we don't need to handle old protocols.  COINJOIN_PROTX_HASH is 70226 and the current is 70230. Any node with a protocol of 70225 would not be connected to.

Reply:
In Java, we still have this code, but it could be removed.